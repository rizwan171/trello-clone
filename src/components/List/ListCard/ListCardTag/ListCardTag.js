import React, { useEffect, useRef } from "react";

const ListCardTag = ({ tag }) => {
  const ref = useRef();

  useEffect(() => {
    ref.current.style.background = tag.colour;
  }, []);

  return (
    <div ref={ref} className="rounded-ibsm p-1">
      {tag.name}
    </div>
  );
};

export default ListCardTag;
