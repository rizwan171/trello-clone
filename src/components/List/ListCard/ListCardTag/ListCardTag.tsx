import { useEffect, useRef } from "react";
import ListCardTagProps from "../../../../types/components/ListCardTagProps";

const ListCardTag: React.FunctionComponent<ListCardTagProps> = ({ tag }) => {
  const ref = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (ref.current) {
      ref.current.style.background = tag.colour;
    }
  }, []);

  return (
    <div ref={ref} className="rounded-ibsm p-1">
      {tag.name}
    </div>
  );
};

export default ListCardTag;
