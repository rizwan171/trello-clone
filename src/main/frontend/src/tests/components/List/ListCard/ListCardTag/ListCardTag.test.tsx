import { render } from "@testing-library/react";
import ListCardTag from "../../../../../components/List/ListCard/ListCardTag/ListCardTag";

describe("ListCardTag", () => {
  it("should render successfully", () => {
    const tag = {
      id: "1",
      name: "Tag 1",
      colour: "#FF3244",
    };

    const view = render(<ListCardTag tag={tag} />);
    expect(view.asFragment()).toMatchSnapshot();
  });
});
