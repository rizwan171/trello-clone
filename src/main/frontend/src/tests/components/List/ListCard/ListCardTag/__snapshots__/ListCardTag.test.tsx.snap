// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCardTag should render successfully 1`] = `
<DocumentFragment>
  <div
    class="rounded-ibsm p-1"
    style="background: rgb(255, 50, 68);"
  >
    Tag 1
  </div>
</DocumentFragment>
`;
